File: C#/1.cs
using System;
using System.Collections.Generic;

class Program {
    static Tuple<int, List<string>> task1(string s) {
        int n = s.Length;
        var steps = new List<string> {};
        string current = "";
        
        for (int i = n / 2; i > 0; i--) {
            if (s.Substring(0, i) == s.Substring(i, i)) {
                for (int j = 0; j < i; j++) {
                    current += s[j];
                    steps.Add(current);
                }
                current += current;
                steps.Add(current);
                current = "";
                for (int j = 2 * i; j < n; j++) {
                    current += s[j];
                    steps.Add(current);
                }
                return Tuple.Create(i + 1 + (n - 2 * i), steps);
            }
        }
        
        foreach (char c in s) {
            current += c;
            steps.Add(current);
        }
        return Tuple.Create(n, steps);
    }
    
    static void Main() {
        string input = Console.ReadLine().Trim();
        var result = task1(input);
        
        Console.Write($"{result.Item1} = ");
        Console.WriteLine(string.Join("->", result.Item2));
    }
}

File: Python/1.py
def task1(s):
    n = len(s)
    steps = []
    current = ""

    for i in range(n // 2, 0, -1):
        if s[:i] == s[i:2*i]:
            for j in range(i):
                current += s[j]
                steps.append(current)
            current += current
            steps.append(current)
            current = ""
            for j in range(2*i, n):
                current += s[j]
                steps.append(current)
            return (i + 1 + (n - 2*i), steps)
    
    for c in s:
        current += c
        steps.append(current)
    return (n, steps)

input_str = input()
result = task1(input_str)

print(f"{result[0]} = ", end="")
print("->".join(result[1]))
